pkg/interfaces/types.go

```diff
@@ -12,21 +12,22 @@ const (
 	StackLabel = "com.docker.stacks.stack_id"
 )
 
-// SwarmStack represents a Stack with all of its elements converted to Engine
+// Stack represents a Stack with all of its elements converted to Engine
 // API types.
-// NOTE: SwarmStack is only used internally for reconciliation of Swarm
+// NOTE: Stack is only used internally for reconciliation of Swarm
 // stacks and is never exposed via the API.
-type SwarmStack struct {
-	ID   string
-	Meta swarm.Meta
-	Spec SwarmStackSpec
+type Stack struct {
+	Version swarm.Version
+	ID      string
+	Meta    swarm.Meta
```

Comment:
This is my mistake, but Meta should actually be an embedded type. That is, the struct should look like this:

```
type Stack struct {
    ID string
    swarm.Meta
    Spec StackSpec
}
```

This keeps with the pattern established with Swarm objects in the existing Docker API

(see here: https://github.com/moby/moby/blob/8d1309222c58143c1e7fa02658e04d06e2e934ce/api/types/swarm/service.go#L5-L13 )